lightkurve/targetpixelfile.py

```diff
@@ -837,10 +848,18 @@ def query_solar_system_objects(self, cadence_mask='outliers', radius=None,
                     aper = 'all'
                 lc = self.to_lightcurve(aperture_mask=aper)
                 cadence_mask = lc.remove_outliers(sigma=sigma, return_mask=True)[1]
-
-            if cadence_mask == 'all':
+                # Avoid searching times with NaN flux; this is necessary because e.g.
+                # `remove_outliers` includes NaNs in its mask.
+                cadence_mask &= ~np.isnan(lc.flux)
```

Comment:
Interesting.  Does the mask returned by `remove_outliers` tend to flag cadences where `lc.flux` is NaN?

Reply:
I honestly don't know. The logic and comments are straight from `LightCurve` variant. Maybe Christina will have a better idea?

Reply:
I'm happy to go with it for now, I don't think it will harm!